routes/runtime/github.rb

```diff
@@ -92,17 +92,17 @@ class CloverRuntime
           fail CloverError.new(400, "InvalidRequest", "No workflow job data available")
         end
 
-        if repository.cache_entries_dataset[key: key, version: version, scope: scope]
-          fail CloverError.new(409, "AlreadyExists", "A cache entry for #{scope} scope already exists with #{key} key and #{version} version.")
-        end
-
         if size && size > GithubRepository::CACHE_SIZE_LIMIT
           fail CloverError.new(400, "InvalidRequest", "The cache size is over the 10GB limit")
         end
 
         entry, upload_id = nil, nil
         DB.transaction do
-          entry = GithubCacheEntry.create_with_id(repository_id: runner.repository.id, key: key, version: version, size: size, scope: scope, created_by: runner.id)
+          begin
+            entry = GithubCacheEntry.create_with_id(repository_id: runner.repository.id, key: key, version: version, size: size, scope: scope, created_by: runner.id)
+          rescue Sequel::ValidationFailed
```

Comment:
`Sequel::ValidationFailed` is a model-level error.  `Sequel::UniqueConstraintViolation` is a database-level error.  You'll get `Sequel::UniqueConstraintViolation` in race conditions, because the validation-level uniqueness check SELECT query won't find an existing row, but an existing row will be added between that check and the INSERT.  If we aren't using pg_auto_constraint_validations in production, we may want to start, if we want to translate the `Sequel::UniqueConstraintViolation` to `Sequel::ValidationFailed`.  Considering the error handling in this case, it doesn't seem like the translation would be helpful, but it wouldn't hurt.

If we want to replicate the `Sequel::UniqueConstraintViolation` in our tests, we'll likely have to add a code injection capability, allowing us to deterministically add the duplicate row between the duplicate check SELECT query and the INSERT query.

Reply:
I added both `Sequel::ValidationFailed` and `Sequel::UniqueConstraintViolation` to the catch clause.